nuclio_sdk/platform.py

```diff
@@ -20,13 +20,31 @@
 
 
 class Platform(object):
-    def __init__(self, kind, namespace="default", connection_provider=None):
+    def __init__(
+        self, kind, namespace="default", connection_provider=None, control_callback=None
+    ):
         self.kind = kind
         self.namespace = namespace
 
         # connection_provider is used for unit testing
         self._connection_provider = connection_provider or http.client.HTTPConnection
 
+        self.control_callback = control_callback
```

Comment:
this needs to be private and not exposed. the implementation is saved to whoever created the platform object